src/Definition/Resolver/EnvironmentVariableResolver.php

```diff
@@ -65,4 +65,9 @@ public function isResolvable(Definition $definition, array $parameters = []) : b
     {
         return true;
     }
+
+    protected function getEnvVariable(string $variableName)
+    {
+        return $_SERVER[$variableName] ?? false;
```

Comment:
After several tests unfortunately it seems that `$_SERVER` is not always populated by PHP. For example on macOS, after installing PHP with Brew, I don't get the variables.

I'm not sure how other frameworks deal with it but I remember having this exact problem with Symfony in the past.

Reply:
@mnapoli You could maybe try checking the php.ini file to see if `variables_order` contains "S" for the super global `$_SERVER`.
Additionally, if `auto_globals_jit` is enabled, PHP.NET says that “the SERVER and ENV variables are created when they're first used (Just In Time) instead of when the script starts.". So if these variables are not used in a script, they will not be initialized.

Reply:
Also, looking at the [EnvVarProcessor](https://github.com/symfony/dependency-injection/blob/c1af27be7ca50969ee4e6cbefb3107715a2070b2/EnvVarProcessor.php#L131-L135) class of Symfony, we can see that it checks, in order, `$_ENV`, `$_SERVER`, `getenv()` and the .env file.

Reply:
I know that this is something that I can configure, but I'm talking about users of PHP-DI.

I want this to work out of the box for everyone, not just me.

Reply:
@mnapoli So instead, as a workaround:
```suggestion
        return $_ENV[$variableName] ?? $_SERVER[$variableName] ?? getenv($variableName);
```
It will suit most users.

Reply:
Yes that sounds like a really good suggestion, and Symfony is doing exactly that!